rs/sns/governance/src/proposal.rs

```diff
@@ -2770,78 +2774,268 @@ mod tests {
         }
     }
 
-    #[test]
-    fn render_upgrade_sns_controlled_canister_proposal() {
+    #[tokio::test]
+    async fn render_upgrade_sns_controlled_canister_proposal() {
         let upgrade = UpgradeSnsControlledCanister {
-            canister_id: Some(basic_principal_id()),
+            canister_id: Some(basic_canister_id()),
             new_canister_wasm: vec![0, 0x61, 0x73, 0x6D, 1, 0, 0, 0],
             canister_upgrade_arg: None,
             mode: Some(CanisterInstallModeProto::Upgrade.into()),
+            chunked_canister_wasm: None,
         };
-        let text = validate_and_render_upgrade_sns_controlled_canister(&upgrade).unwrap();
+        let env = setup_for_upgrade_sns_controlled_canister_tests(&upgrade);
+        let text = validate_and_render_upgrade_sns_controlled_canister(&upgrade, &env)
+            .await
+            .unwrap();
 
         assert_eq!(
             text,
-            r#"# Proposal to upgrade SNS controlled canister:
+            r#"# Proposal to Upgrade an SNS Controlled Canister
 
-## Canister id: bg4sm-wzk
+## Target canister: xbgkv-fyaaa-aaaaa-aaava-cai
 
-## Canister wasm sha256: 93a44bbb96c751218e4c00d479e4c14358122a389acca16205b1e4d0dc5f9476
+## Wasm info
+
+Embedded module with 8 bytes and SHA256 `93a44bbb96c751218e4c00d479e4c14358122a389acca16205b1e4d0dc5f9476`.
 
 ## Mode: Upgrade
 
-## No upgrade arg"#
+## Argument info
+
+No upgrade argument."#
                 .to_string()
         );
     }
 
-    #[test]
-    fn render_upgrade_sns_controlled_canister_proposal_with_upgrade_args() {
+    #[tokio::test]
+    async fn render_upgrade_sns_controlled_canister_proposal_with_chunked_wasm() {
+        let upgrade = UpgradeSnsControlledCanister {
+            canister_id: Some(basic_canister_id()),
+            new_canister_wasm: vec![],
+            canister_upgrade_arg: None,
+            mode: Some(CanisterInstallModeProto::Upgrade.into()),
```

Comment:
no into please.

Reply:
This was a copy and paste, so I'd like to avoid changing this one line while keeping all other occurrences in this file (which are totally orthogonal to the PR). I'll try not to introduce new instances, though.